### PR TITLE
[Batch] Added operators

### PR DESCRIPTION
**Pull request by Briles**
Added conditional operators `& && ||`, the `@` operator,  and the `==` string comparison operator

### CODE REVIEW

File: Batch File/Batch File.sublime-syntax [hunk 1]
```diff
@@ -20,12 +21,21 @@ contexts:
...
+    - match: '@(?=\S)'
+      scope: keyword.operator.at.dosbatch
+    - match: \b(?i:EQU|NEQ|LSS|LEQ|GTR|GEQ|==)\b
+      scope: keyword.operator.comparison.dosbatch
+    - match: '[|]'
       scope: keyword.operator.pipe.dosbatch
-    - match: '&>|\d*>&\d*|\d*(>>|>|<)|\d*<&|\d*<>'
-      scope: keyword.operator.redirect.shell
+    - match: '<&?|>[&>]?'
+      scope: keyword.operator.redirection.dosbatch
```
Comment (Briles) [inline Batch File/Batch File.sublime-syntax]: I removed the handle digits as I didn't feel they belonged as part of the operator

File: Batch File/Batch File.sublime-syntax [hunk 2]
```diff
@@ -20,12 +21,16 @@ contexts:
...
```
Comment (Briles) [inline Batch File/Batch File.sublime-syntax]: I removed the handle digits here as I didn't feel they belonged as part of the operator

File: Batch File/Batch File.sublime-syntax [hunk 3]
```diff
@@ -20,12 +21,18 @@ contexts:
...
+    - match: \b(?i:EQU|NEQ|LSS|LEQ|GTR|GEQ|==)\b
+      scope: keyword.operator.comparison.dosbatch
+    - match: '&&?|\|\|'
+      scope: keyword.operator.logical.dosbatch
+    - match: '\|'
       scope: keyword.operator.pipe.dosbatch
-    - match: '&>|\d*>&\d*|\d*(>>|>|<)|\d*<&|\d*<>'
-      scope: keyword.operator.redirect.shell
+    - match: '<&?|>[&>]?'
+      scope: keyword.operator.redirection.dosbatch
```
Comment (Briles) [inline Batch File/Batch File.sublime-syntax]: I removed the handle digits here as I didn't feel they belonged as part of the operator

File: Batch File/Batch File.sublime-syntax [hunk 4]
```diff
@@ -20,12 +21,18 @@ contexts:
...
-    - match: \b(?:EQU|NEQ|LSS|LEQ|GTR|GEQ)\b
-      scope: keyword.operator.dosbatch
-    - match: "[|]"
+
+  operators:
+    - match: '@(?=\S)'
+      scope: keyword.operator.at.dosbatch
+    - match: \b(?i:EQU|NEQ|LSS|LEQ|GTR|GEQ|==)\b
+      scope: keyword.operator.comparison.dosbatch
+    - match: '&&?|\|\|'
```
Comment (wbond) [inline Batch File/Batch File.sublime-syntax]: Is a single `&` a conditional?

Comment (Briles) [inline Batch File/Batch File.sublime-syntax]: `&` is for running a second command regardless of the outcome of the first. It's equivalent to `;` in bash which is scoped as `keyword.operator.list.shell` in `Shell-Unix-Generic`. Everything I've read<sup><a href="http://www.robvanderwoude.com/condexec.php">1</a></sup> <sup><a href="http://ss64.com/nt/syntax-conditional.html">2</a></sup> <sup><a href="https://www.microsoft.com/resources/documentation/windows/xp/all/proddocs/en-us/ntcmds_shelloverview.mspx?mfr=true">3</a></sup> has associated `&` with the conditional operators.

Comment (wbond) [inline Batch File/Batch File.sublime-syntax]: Ok!

**Pull request merged**